The job **`aarch64-apple`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9293946489/job/25578164152) [(plain)](https://github.com/rust-lang-ci/rust/commit/f839e1da7b318939469aa59243d6f6c81102c711/checks/25578164152/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[TIMING] core::build_steps::tool::ToolBuild { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin, tool: "tidy", path: "src/tools/tidy", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "" } -- 19.214
[TIMING] core::build_steps::tool::Tidy { compiler: Compiler { stage: 0, host: aarch64-apple-darwin }, target: aarch64-apple-darwin } -- 0.000
fmt check
skip untracked path AWSCLIV2.pkg during rustfmt invocations
##[error]Diff in /Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/archive.rs at line 153:
             // that loaded but crashed with an AV upon calling one of the imported
             // functions. Therefore, use binutils to create the import library instead,
             // by writing a .DEF file to the temp dir and calling binutils's dlltool.
-            let def_file_path =
-                tmpdir.join(format!("{lib_name}{name_suffix}.def"));
+            let def_file_path = tmpdir.join(format!("{lib_name}{name_suffix}.def"));
             let def_file_content = format!(
             let def_file_content = format!(
                 "EXPORTS\n{}",
Running `"/Users/runner/work/rust/rust/build/aarch64-apple-darwin/rustfmt/bin/rustfmt" "--config-path" "/Users/runner/work/rust/rust" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/type_of.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/common.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/context.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/builder.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/declare.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/asm.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/abi.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/errors.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/type_.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/value.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/utils.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/metadata/type_map.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/cpp_like.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/native.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_cranelift/patches/bcryptprimitives.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/gdb.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/namespace.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/create_scope_map.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/mono_item.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/allocator.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/base.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/llvm/diagnostic.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/llvm/archive_ro.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/llvm/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/llvm/ffi.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/attributes.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/lib.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/intrinsic.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/llvm_util.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/archive.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/profiling.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/write.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/lto.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/back/owned_target_machine.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/va_arg.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/coverageinfo/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/coverageinfo/ffi.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/coverageinfo/map_data.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/consts.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_llvm/src/callee.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/diagnostic.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/diagnostic_impls.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/translation.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/lock.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/tests.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/emitter.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/json/tests.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/json.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/lib.rs" "/Users/runner/work/rust/rust/compiler/rustc_mir_build/src/lints.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/markdown/parse.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/markdown/mod.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/markdown/tests/parse.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/markdown/tests/term.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/markdown/term.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/error.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/registry.rs" "/Users/runner/work/rust/rust/compiler/rustc_errors/src/styled_buffer.rs" "/Users/runner/work/rust/rust/compiler/rustc_codegen_cranelift/example/std_example.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed May 29 22:32:20 UTC 2024
  network time: Wed, 29 May 2024 22:32:20 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_codegen_llvm/src/back/archive.rs
+++ b/compiler/rustc_codegen_llvm/src/back/archive.rs
@@ -128,11 +128,7 @@ impl ArchiveBuilderBuilder for LlvmArchiveBuilderBuilder {
         is_direct_dependency: bool,
     ) -> PathBuf {
         let name_suffix = if is_direct_dependency { "_imports" } else { "_imports_indirect" };
-        let output_path = {
-            let mut output_path: PathBuf = tmpdir.to_path_buf();
-            output_path.push(format!("{lib_name}{name_suffix}"));
-            output_path.with_extension("lib")
-        };
+        let output_path = tmpdir.join(format!("{lib_name}{name_suffix}.lib"));
 
         let target = &sess.target;
         let mingw_gnu_toolchain = common::is_mingw_gnu_toolchain(target);
@@ -157,8 +153,7 @@ impl ArchiveBuilderBuilder for LlvmArchiveBuilderBuilder {
             // that loaded but crashed with an AV upon calling one of the imported
             // functions. Therefore, use binutils to create the import library instead,
             // by writing a .DEF file to the temp dir and calling binutils's dlltool.
-            let def_file_path =
-                tmpdir.join(format!("{lib_name}{name_suffix}")).with_extension("def");
+            let def_file_path = tmpdir.join(format!("{lib_name}{name_suffix}.def"));
 
             let def_file_content = format!(
                 "EXPORTS\n{}",

--- a/tests/ui/rfcs/rfc-2627-raw-dylib/dlltool-failed.rs
+++ b/tests/ui/rfcs/rfc-2627-raw-dylib/dlltool-failed.rs
@@ -3,8 +3,8 @@
 //@ needs-dlltool
 //@ compile-flags: --crate-type lib --emit link
 //@ normalize-stderr-test: "[^ ']*/dlltool.exe" -> "$$DLLTOOL"
-//@ normalize-stderr-test: "[^ ]*/foo.def" -> "$$DEF_FILE"
-//@ normalize-stderr-test: "[^ ]*/foo.lib" -> "$$LIB_FILE"
+//@ normalize-stderr-test: "[^ ]*/foo.dll_imports.def" -> "$$DEF_FILE"
+//@ normalize-stderr-test: "[^ ]*/foo.dll_imports.lib" -> "$$LIB_FILE"
 //@ normalize-stderr-test: "-m [^ ]*" -> "$$TARGET_MACHINE"
 //@ normalize-stderr-test: "-f [^ ]*" -> "$$ASM_FLAGS"
 //@ normalize-stderr-test: "--temp-prefix [^ ]*/foo.dll" -> "$$TEMP_PREFIX"

